src/Controllers/Common/ContentController.cs

```diff
@@ -793,7 +800,14 @@ public IActionResult GetAvatarByUserId([FromForm] Guid userId)
         List<RaisedPetData> filteredDragons = new List<RaisedPetData>();
         int[] petTypeIDsInt = Array.ConvertAll(petTypeIDs.Split(','), s => int.Parse(s));
         foreach (RaisedPetData dragon in dragons) {
-            if (petTypeIDsInt.Contains(dragon.PetTypeID)) {
+            if (petTypeIDsInt.Contains(dragon.PetTypeID) && (
+                // If api key is not sent,
+                apiKey == null 
+                // or client is not Math Blaster (but api key present),
+                || (ClientVersion.GetVersion(apiKey) & ClientVersion.MB) == 0
```

Comment:
Have you checked whether this change actually breaks something in other games and you need to use version checker? ... The basic assumption is that if the code is needed in X, it is also needed in other games ... until proven otherwise ...

So we do not use version testing "just in case", but only when necessary.

Also based on endpoint name `Unselected` I suppose it should always works this way (not return current/selected pet) and it might fix some problems in other games.

Reply:
Ok, I'll remove the version checking entirely.

Reply:
ok, so remove also `apiKey` from functions argument list